### ISSUE TITLE
Adding audio to images: Correct duration but no audio in final file

### ISSUE THREAD

**Issue by OfficialCRUGG** [Zulko/moviepy#1779]
Hello there,
I am currently working on a project which takes various PNG files and WAV files, joins them together and plays them after each other (with a MP4 transition playing in-between)

As you can see by the comments in the code below, the wav files don't get added as audio. The duration of the PNGs is set correctly though.

The following is my code:
```py
import os
import sys
from moviepy.editor import *

fn = sys.argv[1]
if os.path.exists(fn) and os.path.exists(fn + "/img") and os.path.exists(fn + "/wav"):
    # Clips array
    clips = []

    # Transition Clip
    introClip = VideoFileClip("videos/intro.mp4")

    # Loop through all images (0.png, 1.png, 2.png, ...)
    for i in range(0, len(os.listdir(fn + "/img"))):
        print(str(i))

        # Get the respective wav file (0.wav, 1.wav, 2.wav, ...)
        wav = AudioFileClip(fn + "/wav/" + str(i) + ".wav")

        # Get the image and set the duration to the duration of the wav file
        img = ImageClip(fn + "/img/" + str(i) + ".png").set_duration(wav.duration)

        # Set the audio of the ImageClip to the wav file
        img = img.set_audio(wav)

        # Append the ImageClip and transition to the clips array
        clips.append(img)
        clips.append(introClip)

    # Concatenate all clips
    final = concatenate_videoclips(clips)

    # Save the final video
    final.write_videofile(fn + "/final.mp4", fps=30) # produces file without audio, but with correct durations
```

**Comment by OfficialCRUGG** [comment]
This is a bit embarrassing. Accidentally dragged the file into Chrome and it plays fine there. No audio in QuickTime though. Does anyone know what could cause QuickTime to just not have any audio though?

**Comment by manuelthomsen** [comment]
Yes it's and encoding issue, I force codec="libx264". This works pretty much universally for me.

`finalVideo.write_videofile(
        "Renders/" + globalConfig["render_name"] + ".mp4",
        fps=config["video_fps"],
        preset="ultrafast",
        codec="libx264",
        audio_codec="aac",
        logger=None,
    )`

**Comment by OsaAjani** [comment]
Thank you for your contributions and for reporting issues in this repository. With the release of v2, which introduces significant changes to the codebase and API, we’ve reviewed the backlog of open PRs and issues. Due to the length of the backlog and the likelihood that many of these are either fixed or no longer applicable, we’ve made the decision to close all previous PRs and issues.

If you believe that any of these are still relevant to the current version or if you'd like to reopen a related discussion, please feel free to create a new issue or pull request, referencing the old one.

Thank you for your understanding and continued support!
